### PR TITLE
fix sample sheet validation

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
As described in  #2804, sample sheet validation does not take into account the bcl converter of the flow cell.

### Added

- This PR adds a check inside the method `validate_sample_sheet` of the class `FlowCellDirectoryData` that compares the sample sheet type extracted from the sample sheet itself and the bcl converter of the `FlowCellDirectoryData`. If they don't match, it fails the validation and informs this mismatch in the logs as a warning.

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b fix-default-demultiplex -a
    ```

### How to test

- [x] Try `cg demultiplex all` when a flow cell with an old sample sheet is in the flow cell directory

### Expected test outcome

- [x] Check that the warning message is printed but no error is raised.

## Review

- [x] Tests executed by SD
- [x] "Merge and deploy" approved by CO
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Deploy this branch on 
  - [ ] cg stage
  - [ ] cg production

### CODE REVIEW

File: cg/cli/demultiplex/demux.py [hunk 1]
```diff
@@ -59,7 +59,8 @@ def demultiplex_all(context: CGConfig, flow_cells_directory: click.Path, dry_run
 
         if not flow_cell.validate_sample_sheet():
             LOG.warning(
-                f"Malformed sample sheet. Run cg demultiplex samplesheet validate {flow_cell.sample_sheet_path}",
+                "Malformed sample sheet. "
+                f"Run cg demultiplex sample sheet validate {flow_cell.sample_sheet_path}",
```
Comment (diitaz93) [inline cg/cli/demultiplex/demux.py:63]: this is just to keep the lines within the 100 characters limit

File: cg/models/flow_cell/flow_cell.py [hunk 2]
```diff
@@ -213,6 +221,12 @@ def sample_sheet_exists(self) -> bool:
 
     def validate_sample_sheet(self) -> bool:
         """Validate if sample sheet is on correct format."""
+        sample_type_from_sample_sheet: Type[FlowCellSample] = get_sample_type(
+            self.sample_sheet_path
+        )
+        if BCL_CONVERTER_TO_SAMPLE_MODEL[self.bcl_converter] != sample_type_from_sample_sheet:
+            LOG.warning("Sample sheet type does not match with flow cell bcl converter, aborting")
```
Comment (ChrOertlin) [inline cg/models/flow_cell/flow_cell.py:228]:
```suggestion
            LOG.warning(f"Detected sample sheet type {sample_type_from_sample_sheet} is deprecated. Please generate new sample sheet for the flow cell.")
```

**Review by ChrOertlin — APPROVED** [review]
Looks good.

File: cg/models/flow_cell/flow_cell.py (hunk 2 above)
Comment (diitaz93) [reply cg/models/flow_cell/flow_cell.py:228]: good idea

Comment (diitaz93) [reply cg/models/flow_cell/flow_cell.py]: fixed

**Comment by diitaz93** [comment]
## Test on stage
```shell
Found directory /home/proj/stage/flow_cells/170517_ST-E00266_0210_BHJCFFALXX
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/170517_ST-E00266_0210_BHJCFFALXX
Set flow cell id to BHJCFFALXX
Using dragen for demultiplexing.
Check if demultiplexing is possible for HJCFFALXX
Check if flow cell is ready for downstream processing
Check if sequencing is done
Sequence is done for flow cell HJCFFALXX
Check if copy of data from sequence instrument is ready
All data has been transferred for flow cell HJCFFALXX
Flow cell HJCFFALXX is ready for downstream processing
Check if sample sheet exists
Fetch latest version from bundle HJCFFALXX
Fetching files with tags in [HJCFFALXX,samplesheet]
Fetching files from version 101823
Sample sheet was generated for BCL2FASTQ
Detected bcl2fastq sheet for dragen flow cell. Generate the correct sample sheet or use the correct bcl converter.
Malformed sample sheet. Run cg demultiplex sample sheet validate /home/proj/stage/flow_cells/170517_ST-E00266_0210_BHJCFFALXX/SampleSheet.csv
Found directory /home/proj/stage/flow_cells/190927_A00689_0069_BHLYWYDSXX
```

**Comment by diitaz93** [comment]
## Deployed to stage:
```shell
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/flow_cells/170517_ST-E00266_0210_BHJCFFALXX
done.
Log deploy... done.
cg, version 54.10.4
[js.diazboada@hasta:/home/proj/stage/flow_cells/170517_ST-E00266_0210_BHJCFFALXX] [S_base] 3m23s $ up
```
## Deployed to production:
```shell
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/flow_cells/170517_ST-E00266_0210_BHJCFFALXX
done.
Log deploy... done.
cg, version 54.10.4
[js.diazboada@hasta:/home/proj/stage/flow_cells/170517_ST-E00266_0210_BHJCFFALXX] [P_base] 3m7s $ 
```

**Pull request merged**